MON-2903: add nodeExporter.collectors.systemd settings. (#1892)
openshift/cluster-monitoring-operator: Documentation/api.md

[Back to TOC](#table-of-contents)

## NodeExporterCollectorSystemdConfig

> #### Description

Reviewer: @bburt-rh, can you review this section?

> The `NodeExporterCollectorSystemdConfig` resource works as an on/off switch for the `systemd` collector of the `node-exporter` agent. By default, the `systemd` collector is disabled. If enabled, the following metrics become available: `node_systemd_socket_accepted_connections_total`, `node_systemd_socket_current_connections`, `node_systemd_socket_refused_connections_total`, `node_systemd_system_running`, `node_systemd_timer_last_trigger_seconds`, `node_systemd_units`, `node_systemd_version`. We can also use the `units` parameter to choose the systemd units to be included by the `systemd` collector. The selected units will generate the `node_systemd_unit_state` metric, which shows the state of each systemd unit. However, its cardinality might be high (5 series per unit per node). If you enable this collector with a long list of selected units, closely monitor the `prometheus-k8s` deployment for excessive memory usage.

Reviewer 2: ```suggestion
The `NodeExporterCollectorSystemdConfig` resource works as an on/off switch for the `systemd` collector of the `node-exporter` agent. By default, the `systemd` collector is disabled. If enabled, the following metrics become available: `node_systemd_socket_accepted_connections_total`, `node_systemd_socket_current_connections`, `node_systemd_socket_refused_connections_total`, `node_systemd_system_running`, `node_systemd_timer_last_trigger_seconds`, `node_systemd_units`, `node_systemd_version`. You can also use the `units` parameter to select the systemd units to be included by the `systemd` collector. The selected units are used to generate the `node_systemd_unit_state` metric, which shows the state of each systemd unit. However, this metric's cardinality might be high (5 series per unit per node). If you enable this collector with a long list of selected units, closely monitor the `prometheus-k8s` deployment for excessive memory usage.
```

<em>appears in: [NodeExporterCollectorConfig](#nodeexportercollectorconfig)</em>

| Property | Type | Description |
| -------- | ---- | ----------- |
| enabled | bool | A Boolean flag that enables or disables the `systemd` collector. |

> | units | []string | A list of regex patterns that match systemd units to be included by the `systemd` collector. The default value is `^$`, which means that no systemd units are included. |

Reviewer 2: ```suggestion
| units | []string | A list of regular expression (regex) patterns that match systemd units to be included by the `systemd` collector. The default value is `^$`, which means that no systemd units are included. |
```

Reviewer 2: Per Red Hat style guidance, `regex` has to always be spelled out.
